general/src/protocol/contexts.rs

```diff
@@ -525,6 +525,33 @@ impl FromValue for TraceStatus {
                     Annotated(None, meta)
                 }
             },
+            Annotated(Some(Value::I64(value)), mut meta) => Annotated(
+                Some(match value {
+                    0 => SpanStatus::Ok,
```

Comment:
You could cast this from a u8 with bounds checks, if you’d like to avoid the duplication. Not sure if we do this in `general` already, though.